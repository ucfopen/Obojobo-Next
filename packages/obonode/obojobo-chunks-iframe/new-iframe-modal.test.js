import React from 'react'
import { mount } from 'enzyme'
import TestRenderer, { act } from 'react-test-renderer'
import NewIFrameModal from './new-iframe-modal'
import IFrameContentTypes from './iframe-content-types'

jest.mock('use-debounce', () => ({
	useDebouncedCallback: cb => {
		const fn = () => {
			cb()
		}

		return fn
	}
}))

const testRendererOptions = {
	createNodeMock: element => {
		if (element.type === 'input') {
			return {
				focus: () => {},
				select: () => {}
			}
		}
	}
}

describe('NewIFrameModal', () => {
	beforeEach(() => {
		jest.clearAllMocks()
	})

	test('NewIFrameModal component mounts as expected', () => {
		const component = mount(
			<NewIFrameModal
				content={{
					src: '',
					contentType: IFrameContentTypes.MEDIA
				}}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('NewIFrameModal component calls onConfirm from props', () => {
		const onConfirm = jest.fn()

		const component = mount(
			<NewIFrameModal
				content={{
					src: '',
					contentType: IFrameContentTypes.MEDIA
				}}
				onConfirm={onConfirm}
			/>
		)

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).toHaveBeenCalled()
	})

	test('NewIFrameModal component focuses on first element', () => {
		const component = TestRenderer.create(
			<NewIFrameModal
				content={{
					src: '',
					contentType: IFrameContentTypes.MEDIA
				}}
			/>,
			testRendererOptions
		)

		expect(component.toJSON()).toMatchSnapshot()
	})

	test('NewIFrameModal component changes src', () => {
		const component = TestRenderer.create(
			<NewIFrameModal
				content={{
					src: '',
					contentType: IFrameContentTypes.MEDIA
				}}
			/>,
			testRendererOptions
		)

		const startState = component.toJSON()
		const input = component.root.findAllByType('input')[1]
		act(() => {
			input.props.onChange({ target: { value: 'https://' } })
		})

		const endState = component.toJSON()
		expect(endState).toMatchSnapshot()
		expect(startState).not.toEqual(endState)
	})

	test('NewIFrameModal component correctly detects content type', () => {
		const component = TestRenderer.create(
			<NewIFrameModal
				content={{
					src: '',
					contentType: IFrameContentTypes.MEDIA
				}}
			/>,
			testRendererOptions
		)
		const startState = component.toJSON()
		const input = component.root.findAllByType('input')[1]

		// Changing the src input to trigger a content type detection (Detection: Media)
		act(() => {
			input.props.onChange({ target: { value: "<iframe src='https://www'></iframe>" } })
		})

		let endState = component.toJSON()
		expect(endState).toMatchSnapshot()
		expect(startState).not.toEqual(endState)

		// Changing the src input to trigger a content type detection (Detection: Webpage)
		act(() => {
			input.props.onChange({ target: { value: 'https://www' } })
		})

		endState = component.toJSON()
		expect(endState).toMatchSnapshot()
		expect(startState).not.toEqual(endState)
	})

	test('NewIFrameModal component opens section explaining why the iframe preview is not working', () => {
		const component = mount(
			<NewIFrameModal
				content={{
					src: 'https://',
					contentType: IFrameContentTypes.WEBPAGE
				}}
			/>
		)

		component
			.find('button')
			.at(0)
			.simulate('click')
		expect(component.html()).toMatchSnapshot()
	})

	test('NewIFrameModal shows message after invalid src is typed', () => {
		const component = TestRenderer.create(
			<NewIFrameModal
				content={{
					src: 'mock-src',
					contentType: IFrameContentTypes.WEBPAGE
				}}
			/>,
			testRendererOptions
		)

		const startState = component.toJSON()
		const input = component.root.findAllByType('input')[1]

		// Changing empty src input to invalid embed code or url
		act(() => {
			input.props.onChange({ target: { value: 'mock-invalid' } })
		})

		const endState = component.toJSON()
		expect(endState).toMatchSnapshot()
		expect(startState).not.toEqual(endState)
	})
})
