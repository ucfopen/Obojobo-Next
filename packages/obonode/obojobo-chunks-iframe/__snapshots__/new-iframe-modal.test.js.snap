// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewIFrameModal NewIFrameModal component changes src 1`] = `
<div
  className="obojobo-draft--components--modal--simple-dialog"
>
  <div
    className="obojobo-draft--components--modal--dialog "
    style={null}
  >
    <div
      aria-labelledby="obojobo-draft--components--modal--modal--content"
      className="obojobo-draft--components--modal--modal"
      role="dialog"
      tabIndex="-1"
    >
      <input
        aria-label="Start of dialog"
        className="first-tab"
        onFocus={[Function]}
        type="text"
      />
      <div
        className="content"
        id="obojobo-draft--components--modal--modal--content"
      >
        <h1
          className="heading"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          New Embedded IFrame
        </h1>
        <div
          className="dialog-content"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <div
            className="new-iframe-modal"
          >
            <header>
              <p>
                Paste either an iframe embed code or a URL to embed:
              </p>
              <input
                onChange={[Function]}
                placeholder="<iframe src=\\"https://example.com\\"/> or \\"https://example.com\\""
                type="text"
                value=""
              />
            </header>
            <div
              className="preview"
            >
              <p>
                Embedded preview:
              </p>
              <div
                className="no-preview"
              >
                <span>
                  Paste a link or embed code above to see the preview
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="controls"
        >
          <div
            className="obojobo-draft--components--button alt-action is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
          </div>
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[Function]}
              type="button"
            >
              Preview is good - Continue...
            </button>
          </div>
        </div>
      </div>
      <input
        aria-label="End of dialog"
        className="last-tab"
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`NewIFrameModal NewIFrameModal component correctly detects content type 1`] = `
<div
  className="obojobo-draft--components--modal--simple-dialog"
>
  <div
    className="obojobo-draft--components--modal--dialog "
    style={null}
  >
    <div
      aria-labelledby="obojobo-draft--components--modal--modal--content"
      className="obojobo-draft--components--modal--modal"
      role="dialog"
      tabIndex="-1"
    >
      <input
        aria-label="Start of dialog"
        className="first-tab"
        onFocus={[Function]}
        type="text"
      />
      <div
        className="content"
        id="obojobo-draft--components--modal--modal--content"
      >
        <h1
          className="heading"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          New Embedded IFrame
        </h1>
        <div
          className="dialog-content"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <div
            className="new-iframe-modal"
          >
            <header>
              <p>
                Paste either an iframe embed code or a URL to embed:
              </p>
              <input
                onChange={[Function]}
                placeholder="<iframe src=\\"https://example.com\\"/> or \\"https://example.com\\""
                type="text"
                value=""
              />
            </header>
            <div
              className="preview"
            >
              <p>
                Embedded preview:
              </p>
              <div
                className="no-preview"
              >
                <span>
                  Paste a link or embed code above to see the preview
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="controls"
        >
          <div
            className="obojobo-draft--components--button alt-action is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
          </div>
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[Function]}
              type="button"
            >
              Preview is good - Continue...
            </button>
          </div>
        </div>
      </div>
      <input
        aria-label="End of dialog"
        className="last-tab"
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`NewIFrameModal NewIFrameModal component correctly detects content type 2`] = `
<div
  className="obojobo-draft--components--modal--simple-dialog"
>
  <div
    className="obojobo-draft--components--modal--dialog "
    style={null}
  >
    <div
      aria-labelledby="obojobo-draft--components--modal--modal--content"
      className="obojobo-draft--components--modal--modal"
      role="dialog"
      tabIndex="-1"
    >
      <input
        aria-label="Start of dialog"
        className="first-tab"
        onFocus={[Function]}
        type="text"
      />
      <div
        className="content"
        id="obojobo-draft--components--modal--modal--content"
      >
        <h1
          className="heading"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          New Embedded IFrame
        </h1>
        <div
          className="dialog-content"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <div
            className="new-iframe-modal"
          >
            <header>
              <p>
                Paste either an iframe embed code or a URL to embed:
              </p>
              <input
                onChange={[Function]}
                placeholder="<iframe src=\\"https://example.com\\"/> or \\"https://example.com\\""
                type="text"
                value=""
              />
            </header>
            <div
              className="preview"
            >
              <p>
                Embedded preview:
              </p>
              <div
                className="no-preview"
              >
                <span>
                  Paste a link or embed code above to see the preview
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="controls"
        >
          <div
            className="obojobo-draft--components--button alt-action is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
          </div>
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[Function]}
              type="button"
            >
              Preview is good - Continue...
            </button>
          </div>
        </div>
      </div>
      <input
        aria-label="End of dialog"
        className="last-tab"
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`NewIFrameModal NewIFrameModal component focuses on first element 1`] = `
<div
  className="obojobo-draft--components--modal--simple-dialog"
>
  <div
    className="obojobo-draft--components--modal--dialog "
    style={null}
  >
    <div
      aria-labelledby="obojobo-draft--components--modal--modal--content"
      className="obojobo-draft--components--modal--modal"
      role="dialog"
      tabIndex="-1"
    >
      <input
        aria-label="Start of dialog"
        className="first-tab"
        onFocus={[Function]}
        type="text"
      />
      <div
        className="content"
        id="obojobo-draft--components--modal--modal--content"
      >
        <h1
          className="heading"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          New Embedded IFrame
        </h1>
        <div
          className="dialog-content"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <div
            className="new-iframe-modal"
          >
            <header>
              <p>
                Paste either an iframe embed code or a URL to embed:
              </p>
              <input
                onChange={[Function]}
                placeholder="<iframe src=\\"https://example.com\\"/> or \\"https://example.com\\""
                type="text"
                value=""
              />
            </header>
            <div
              className="preview"
            >
              <p>
                Embedded preview:
              </p>
              <div
                className="no-preview"
              >
                <span>
                  Paste a link or embed code above to see the preview
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="controls"
        >
          <div
            className="obojobo-draft--components--button alt-action is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
          </div>
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[Function]}
              type="button"
            >
              Preview is good - Continue...
            </button>
          </div>
        </div>
      </div>
      <input
        aria-label="End of dialog"
        className="last-tab"
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`NewIFrameModal NewIFrameModal component mounts as expected 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\" tabindex=\\"-1\\"><input aria-label=\\"Start of dialog\\" class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">New Embedded IFrame</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"new-iframe-modal\\"><header><p>Paste either an iframe embed code or a URL to embed:</p><input type=\\"text\\" placeholder=\\"<iframe src=&quot;https://example.com&quot;/> or &quot;https://example.com&quot;\\" value=\\"\\"></header><div class=\\"preview\\"><p>Embedded preview:</p><div class=\\"no-preview\\"><span>Paste a link or embed code above to see the preview</span></div></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\" contenteditable=\\"false\\"><button type=\\"button\\" class=\\"button\\" contenteditable=\\"false\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button type=\\"button\\" class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">Preview is good - Continue...</button></div></div></div><input aria-label=\\"End of dialog\\" class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`NewIFrameModal NewIFrameModal component opens section explaining why the iframe preview is not working 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\" tabindex=\\"-1\\"><input aria-label=\\"Start of dialog\\" class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">New Embedded IFrame</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"new-iframe-modal\\"><header><p>Paste either an iframe embed code or a URL to embed:</p><input type=\\"text\\" placeholder=\\"<iframe src=&quot;https://example.com&quot;/> or &quot;https://example.com&quot;\\" value=\\"https://\\"></header><div class=\\"preview\\"><p>Embedded preview:</p><div class=\\"preview-with-iframe\\"><iframe src=\\"\\"></iframe><section><p>Does the preview look good?</p><div class=\\"preview-not-working\\">If the preview above is not what you expected, keep in mind that some pages inside your IFrame may restrict their content, thus not allowing them to be shown within Obojobo. Also, if you are trying to embed media instead of an IFrame, make sure to paste your IFrame's embed code (starting with &lt;iframe...) and not only the regular URL.</div></section></div></div></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\" contenteditable=\\"false\\"><button type=\\"button\\" class=\\"button\\" contenteditable=\\"false\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button type=\\"button\\" class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">Preview is good - Continue...</button></div></div></div><input aria-label=\\"End of dialog\\" class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`NewIFrameModal NewIFrameModal shows message after invalid src is typed 1`] = `
<div
  className="obojobo-draft--components--modal--simple-dialog"
>
  <div
    className="obojobo-draft--components--modal--dialog "
    style={null}
  >
    <div
      aria-labelledby="obojobo-draft--components--modal--modal--content"
      className="obojobo-draft--components--modal--modal"
      role="dialog"
      tabIndex="-1"
    >
      <input
        aria-label="Start of dialog"
        className="first-tab"
        onFocus={[Function]}
        type="text"
      />
      <div
        className="content"
        id="obojobo-draft--components--modal--modal--content"
      >
        <h1
          className="heading"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          New Embedded IFrame
        </h1>
        <div
          className="dialog-content"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <div
            className="new-iframe-modal"
          >
            <header>
              <p>
                Paste either an iframe embed code or a URL to embed:
              </p>
              <input
                onChange={[Function]}
                placeholder="<iframe src=\\"https://example.com\\"/> or \\"https://example.com\\""
                type="text"
                value="mock-src"
              />
            </header>
            <div
              className="preview"
            >
              <p>
                Embedded preview:
              </p>
              <div
                className="no-preview"
              >
                <span>
                  Make sure you are using a valid URL or an IFrame embed code
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="controls"
        >
          <div
            className="obojobo-draft--components--button alt-action is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
          </div>
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[Function]}
              type="button"
            >
              Preview is good - Continue...
            </button>
          </div>
        </div>
      </div>
      <input
        aria-label="End of dialog"
        className="last-tab"
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;
